src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs

```diff
@@ -348,6 +648,101 @@ private static double[] ComputeSequenceDeltas(IReadOnlyList<double> a, IReadOnly
             return delta;
         }
 
+        private static ImmutableDictionary<string, TResult>
+            PermutationFeatureImportance<TMetric, TResult>(
+                IHostEnvironment env,
+                ITransformer model,
+                IDataView data,
+                Func<TResult> resultInitializerParam,
+                Func<TMetric, TMetric, TMetric> deltaFuncParam,
+                Func<IDataView, TMetric> evaluationFuncParam,
+                bool useFeatureWeightFilter,
+                int? numberOfExamplesToUse,
+                int permutationCount) where TResult : IMetricsStatistics<TMetric>
+        {
+            env.CheckValue(data, nameof(data));
+            env.CheckValue(model, nameof(model));
+
+            ITransformer lastTransformer = null;
+
+            if (model is TransformerChain<ITransformer> chain)
+            {
+                foreach (var transformer in chain.Reverse())
+                {
+                    if (transformer is ISingleFeaturePredictionTransformer)
+                    {
+                        lastTransformer = transformer;
+                        break;
+                    }
+                }
+            }
+            else lastTransformer = model as ISingleFeaturePredictionTransformer;
+
+            env.CheckValue(lastTransformer, nameof(lastTransformer));
```

Comment:
We probably need a better exception here to let people know what the problem is. Just throwing an `ArgumentNullException("lastTransformer")` doesn't really help someone know what the issue is when they have this wrong.

Reply:
Good point. Does something like:
"The model provided does not have a compatible predictor" sound good to you?

Reply:
For now I have updated it to that. If you think it needs to be more specific let know.